sbom.py

```diff
@@ -90,9 +90,19 @@ class CreationInfo(TypedDict):
     licenseListVersion: str
 
 
-def spdx_id(value: str) -> str:
+# Cache of values that we've seen already. We use this
+# to de-duplicate values and their corresponding SPDX ID.
+_SPDX_IDS_TO_VALUES = {}
+
+
+def spdx_id(value: LiteralString) -> str:
     """Encode a value into characters that are valid in an SPDX ID"""
-    return re.sub(r"[^a-zA-Z0-9.\-]+", "-", value)
+    spdx_id = re.sub(r"[^a-zA-Z0-9.\-]+", "-", value)
```

Comment:
This `spdx_id` shadows the `spdx_id` function name, can we have unique names to remove the IDE warning squiggle?